v-next/hardhat-utils/src/internal/bytecode.ts

```diff
@@ -0,0 +1,155 @@
+import type { PrefixedHexString } from "../hex.js";
+
+import {
+  AmbiguousLibraryNameError,
+  InvalidLibraryAddressError,
+  MissingLibrariesError,
+  OverlappingLibrariesError,
+  UnnecessaryLibraryError,
+} from "../errors/bytecode.js";
+import { isAddress } from "../eth.js";
+
+export interface Artifact {
```

Comment:
How do we think about types within the hh-utils?

I was tempted to suggest that we should be leveraging a top level `Artifact` type, but that would imply a new package that the utils could depend on that is shared with hardhat.

So would a fair description of our rule be, that these are the minimal types that describe the util functions signature and we will take the pain of keeping them in sync with any core types (e.g. core `Artifact`)?

Reply:
That’s correct. Types in hh-utils should remain minimal, used internally, and not exposed in the public API. Unfortunately, this means we’ll need to manually keep them in sync with the types in Hardhat or plugins.